### PR TITLE
adding cli challenge to student exercises part 4

### LINKED ISSUE #105: Add challenges to Student Exercises part 4
Since some students finish all orientation exercises much earlier than others, we should add some challenges to the final exercise that reinforce the lessons from orientation. Students who have completed all other exercises can then work on these challenges.

### PR DESCRIPTION
**Pull request by askingalot**
resolves #105

### CODE REVIEW

File: book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md [hunk 1]
```diff
@@ -5,10 +5,77 @@
...
-1. Fnd all the exercises in the database where the language is JavaScript.
-1. Insert a new exercise into the database.
-1. Find all instructors in the database. Include each instructor's cohort.
-1. Insert a new instructor into the database. Assign the instructor to an existing cohort.
-1. Assign an existing exercise to an existing student.
-1. **Challenge** - Find all the students in the database. Include each student's cohort AND each student's list of exercises.
\ No newline at end of file
+1. Write the necessary C# code in `Repository.cs` and `Program.cs` to perform the following actions. Make sure to print results of each action to the console.
+    1. Query the database for all the Exercises.
+    1. Fnd all the exercises in the database where the language is JavaScript.
```
Comment (stevebrownlee) [inline book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md]: `Fnd` misspelled.

File: book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md [hunk 2]
```diff
@@ -5,10 +5,77 @@
...
+    1. Insert a new exercise into the database.
+    1. Find all instructors in the database. Include each instructor's cohort.
+    1. Insert a new instructor into the database. Assign the instructor to an existing cohort.
+    1. Assign an existing exercise to an existing student.
+    1. **Challenge**: Find all the students in the database. Include each student's cohort AND each student's list of exercises.
```
Comment (stevebrownlee) [inline book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md]: Style these as other challenges in the course are 

```
## Challenge

List all the students in the database. Include each student's cohort AND each student's list of exercises.
```

File: book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md [hunk 3]
```diff
@@ -5,10 +5,77 @@
...
+    1. **Challenge**: Write a method in the `Repository` class that accepts an `Exercise` and a `Cohort` and assigns that exercise to each student in the cohort IF and ONLY IF the student has not already been assigned the exercise.
+
+### Further Challenge
```
Comment (stevebrownlee) [inline book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md]: Mark this as an advanced challenge

```
## Advanced Challenge

> **NOTE**: _Only work on this challenge if you've completed ALL the other exercises assigned during Orientation._
```

File: book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md [hunk 4]
```diff
@@ -5,10 +5,77 @@
...
+>**NOTE**: _Only work on this challenge if you've completed ALL the other exercises assigned during Orientation._
+1. Modify your program to create a menu and accept input from the user using the [Console.ReadLine()](https://docs.microsoft.com/en-us/dotnet/api/system.console.readline?redirectedfrom=MSDN&view=netframework-4.7.2#System_Console_ReadLine) method.
+    User the following program as an example for creating a menu.
```
Comment (stevebrownlee) [inline book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md]: `Use the following`

File: book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md [hunk 5]
```diff
@@ -5,10 +5,77 @@
...
\ No newline at end of file
+1. Write the necessary C# code in `Repository.cs` and `Program.cs` to perform the following actions. Make sure to print results of each action to the console.
+    1. Query the database for all the Exercises.
+    1. Fnd all the exercises in the database where the language is JavaScript.
+    1. Insert a new exercise into the database.
+    1. Find all instructors in the database. Include each instructor's cohort.
+    1. Insert a new instructor into the database. Assign the instructor to an existing cohort.
+    1. Assign an existing exercise to an existing student.
+    1. **Challenge**: Find all the students in the database. Include each student's cohort AND each student's list of exercises.
+    1. **Challenge**: Write a method in the `Repository` class that accepts an `Exercise` and a `Cohort` and assigns that exercise to each student in the cohort IF and ONLY IF the student has not already been assigned the exercise.
```
Comment (stevebrownlee) [inline book-1-orientation/chapters/STUDENT_EXERCISES_ADONET.md]: Style these as other challenges in the course are

```
## Challenge

Write a method in the `Repository` class that...
```

**Comment by copypastedeveloper** [comment]
I'd take out the example if it's a challenge.  The example largely allows them to copy place and replace to make it work.  Maybe suggest the while loop since that could be hard to get on their own.

**Comment by askingalot** [comment]
I haven’t shown the students in c29 how to get user input, nor the while(true) loop, so I felt like giving them that example was appropriate.

The challenge is about how to write the sql needed to perform the various menu actions, to think about what belongs in the repository versus what belongs in Program.cs, plus having them decide how to get enough information from the user to fulfil each menu action. 

I feel like the example program will prevent them from spinning their wheels trying to figure out less important things, and give them more time to focus on the objectives of the exercise.

**Comment by copypastedeveloper** [comment]
I see.  I guess I missed that those prompts all required writing sql that didn't exist to date.  That seems reasonable.

**Comment by stevebrownlee** [comment]
Yeah @copypastedeveloper this is just a scaled-down version of what the CLI/TDD sprint used to be. Since students would be doing this challenge in place of, or more likely beforehand, that group project, they need to be given somewhere to start.

**Pull request merged**